presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/CatalogAccessControlRule.java

```diff
@@ -15,35 +15,90 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Maps;
 
+import java.util.Locale;
+import java.util.Map;
 import java.util.Optional;
 import java.util.regex.Pattern;
 
 import static java.util.Objects.requireNonNull;
 
 public class CatalogAccessControlRule
 {
-    private final boolean allow;
+    private final AccessMode accessMode;
     private final Optional<Pattern> userRegex;
     private final Optional<Pattern> catalogRegex;
 
     @JsonCreator
     public CatalogAccessControlRule(
-            @JsonProperty("allow") boolean allow,
+            @JsonProperty("allow") AccessMode accessMode,
             @JsonProperty("user") Optional<Pattern> userRegex,
             @JsonProperty("catalog") Optional<Pattern> catalogRegex)
     {
-        this.allow = allow;
+        this.accessMode = requireNonNull(accessMode, "accessMode is null");
         this.userRegex = requireNonNull(userRegex, "userRegex is null");
         this.catalogRegex = requireNonNull(catalogRegex, "catalogRegex is null");
     }
 
-    public Optional<Boolean> match(String user, String catalog)
+    public Optional<AccessMode> match(String user, String catalog)
     {
         if (userRegex.map(regex -> regex.matcher(user).matches()).orElse(true) &&
                 catalogRegex.map(regex -> regex.matcher(catalog).matches()).orElse(true)) {
-            return Optional.of(allow);
+            return Optional.of(accessMode);
         }
         return Optional.empty();
     }
+
+    public enum AccessMode
+    {
+        ALL("all"),
+        READ_ONLY("read-only"),
+        NONE("none");
+
+        private static final Map<String, AccessMode> modeByName = Maps.uniqueIndex(ImmutableList.copyOf(AccessMode.values()), AccessMode::toString);
+
+        private final String stringValue;
+
+        AccessMode(String stringValue)
+        {
+            this.stringValue = requireNonNull(stringValue, "stringValue is null");
+        }
+
+        @JsonValue
+        @Override
+        public String toString()
```

Comment:
I'd prefer not to override `toString()` (nor `name()`) and have a dedicated method for this. Maybe `getStringValue()`.

@electrum WDYT?